Userland/Libraries/LibWeb/Painting/ImagePaintable.cpp

```diff
@@ -62,11 +62,36 @@ void ImagePaintable::paint(PaintContext& context, PaintPhase phase) const
 
     PaintableBox::paint(context, phase);
 
+    auto paint_alt_rect = [&](Gfx::Rect<int> rect, Palette palette) {
+        Gfx::Path top_left;
+        Gfx::Path bottom_right;
+        Gfx::IntRect shrunked = rect.shrunken(2, 2);
+
+        top_left.move_to(rect.bottom_left().to_type<float>());
+        top_left.line_to(rect.top_left().to_type<float>());
+        top_left.line_to(rect.top_right().to_type<float>());
+        top_left.line_to(shrunked.top_right().to_type<float>());
+        top_left.line_to(shrunked.top_left().to_type<float>());
+        top_left.line_to(shrunked.bottom_left().to_type<float>());
+        top_left.move_to(rect.bottom_left().to_type<float>());
+
+        context.recording_painter().fill_path({ .path = top_left, .color = palette.threed_shadow1() });
+
+        bottom_right.move_to(rect.bottom_left().to_type<float>());
+        bottom_right.line_to(shrunked.bottom_left().to_type<float>());
+        bottom_right.line_to(shrunked.bottom_right().to_type<float>());
+        bottom_right.line_to(shrunked.top_right().to_type<float>());
+        bottom_right.line_to(rect.top_right().to_type<float>());
+        bottom_right.line_to(rect.bottom_right().to_type<float>());
+
+        context.recording_painter().fill_path({ .path = bottom_right, .color = palette.button() });
+    };
+
     if (phase == PaintPhase::Foreground) {
         auto image_rect = context.rounded_device_rect(absolute_rect());
         if (m_renders_as_alt_text) {
             auto enclosing_rect = context.enclosing_device_rect(absolute_rect()).to_type<int>();
-            context.recording_painter().paint_frame(enclosing_rect, context.palette(), Gfx::FrameStyle::SunkenContainer);
+            paint_alt_rect(enclosing_rect, context.palette());
```

Comment:
Refactoring this slightly so that you pass a `Gfx::FloatRect` into `paint_alt_rect()` would avoid any possible loss of precision here. It also means that you won't need to do all those `to_type<float>()` calls above.